From: Christian Borntraeger <borntraeger@de.ibm.com>
Subject: kernel/kvm: Fix MSA3/MSA4 detection
Patch-mainline: no (SLES only)
References: bsc#936916, LTC#127868

Description:  kernel/kvm: Fix MSA3/MSA4 detection
Symptom:      Applications that use MSA3/MSA4 crash under KVM
Problem:      The kvm tech preview announces the facility bits for
              MSA3/MSA4 to its guest, but the implementation is
              missing.
Solution:     Remove MSA3/MSA4 bits from the stfle mask.
Reproduction: Install IBM Java 8

Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/kvm/kvm-s390.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -1156,7 +1156,7 @@ static int __init kvm_s390_init(void)
 	}
 	memcpy(vfacilities, S390_lowcore.stfle_fac_list, 16);
 	vfacilities[0] &= 0xff82fff3f47c0000UL;
-	vfacilities[1] &= 0x001c000000000000UL;
+	vfacilities[1] &= 0x0010000000000000UL;
 	return 0;
 }
 
